.github/workflows/codeql.yaml

```diff
@@ -0,0 +1,35 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
```

Comment:
is it going to add too much extra time to PR validation?

Reply:
Validated it here: https://github.com/aramase/secrets-store-csi-driver-provider-azure/runs/8259861468?check_suite_focus=true and it only takes 3m40s